cumulusci/tasks/metadeploy.py

```diff
@@ -175,7 +176,9 @@ def _publish_plan(self, product, version, plan_name, plan_config, steps):
             product, plan_name, plan_config
         )
 
-        parsed_plan = Plan.parse_obj(self.project_config.config["plans"][plan_name])
+        order_key = plan_config.pop("order_key")
+        parsed_plan = Plan.parse_obj(plan_config)
+        # TODO: clean this up to not use both Plan and MetaDeployPlan
```

Comment:
💭 Yeah, the interaction between these two needs to be revisited. Maybe a class method on `MetaDeployPlan` that'll do the data munging or something.